### PR TITLE
DEPR: Series/DataFrame with tzaware data and tznaive dtype

### PR DESCRIPTION
**Pull request by jbrockmendel**
- [ ] closes #xxxx
- [x] tests added / passed
- [x] Ensure all linting tests pass, see [here](https://pandas.pydata.org/pandas-docs/dev/development/contributing.html#code-standards) for how to run them
- [x] whatsnew entry

### CODE REVIEW

**Comment by simonjayhawkins** [comment]
@jbrockmendel can you resolve conflicts

**Comment by simonjayhawkins** [comment]
and add a release note. and a link to where this has been discussed.

**Comment by jbrockmendel** [comment]
whatsnew added + green

File: pandas/core/dtypes/cast.py [hunk 1]
```diff
@@ -219,6 +219,8 @@ def maybe_unbox_datetimelike(value: Scalar, dtype: DtypeObj) -> Scalar:
     elif isinstance(value, Timestamp):
         if value.tz is None:
             value = value.to_datetime64()
+        elif not isinstance(dtype, DatetimeTZDtype):
+            raise TypeError("Cannot unbox tzaware Timestamp to tznaive dtype")
```
Comment (jreback) [inline pandas/core/dtypes/cast.py:223]: this hit in tests?

File: pandas/core/dtypes/cast.py [hunk 2]
```diff
@@ -1873,14 +1887,48 @@ def construct_1d_arraylike_from_scalar(
...
+    """
+    Wrap maybe_unbox_datetimelike with a check for a timezone-aware Timestamp
+    along with a timezone-naive datetime64 dtype, which is deprecated.
+    """
+    # Caller is responsible for checking dtype.kind in ["m", "M"]
+    try:
+        value = maybe_unbox_datetimelike(value, dtype)
+    except TypeError:
+        if (
+            isinstance(value, Timestamp)
```
Comment (jreback) [inline pandas/core/dtypes/cast.py:1910]: does this apply to a datetime as well?

File: pandas/tests/frame/test_constructors.py [hunk 3]
```diff
@@ -2436,12 +2447,41 @@ def test_construction_preserves_tzaware_dtypes(self, tz):
...
-        expected = DataFrame({"d": [Timestamp("2019")]})
+        ts = Timestamp("2019", tz=tz)
+        ts_naive = Timestamp("2019")
+
+        with tm.assert_produces_warning(FutureWarning):
+            result = DataFrame({0: [ts]}, dtype="datetime64[ns]")
+
+        expected = DataFrame({0: [ts_naive]})
+        tm.assert_frame_equal(result, expected)
+
```
Comment (jreback) [inline pandas/tests/frame/test_constructors.py:2460]: maybe worth parameterzing?

**Review by jreback — CHANGES_REQUESTED** [review]

File: pandas/core/dtypes/cast.py (hunk 1 above)
Comment (jbrockmendel) [reply pandas/core/dtypes/cast.py:223]: at least 37 of em

File: pandas/core/dtypes/cast.py (hunk 2 above)
Comment (jbrockmendel) [reply pandas/core/dtypes/cast.py:1910]: good catch, will update, along with test

File: pandas/tests/frame/test_constructors.py (hunk 3 above)
Comment (jbrockmendel) [reply pandas/tests/frame/test_constructors.py:2460]: i think theres a decent chance these can end up in the TestFromScalar class, will need to see how much hassle it is

**Review by jreback — APPROVED** [review]

**Pull request merged**